packages/lit-html/README.md

```diff
@@ -1,14 +1,133 @@
 [![Mentioned in Awesome lit-html](https://awesome.re/mentioned-badge.svg)](https://github.com/web-padawan/awesome-lit-html)
 
-## Experimental lit-html 2.0
+## 🚨 About this pre-release
+
+This is a major version pre-release of lit-html 2.0. See issue
+[#1182](https://github.com/Polymer/lit-html/issues/1182) for the full list of changes
+planned/considered for this release.
+
+This pre-release is not yet feature complete or API stable. Please note the
+breaking changes, known issues, and limitations below, and use at your risk
+until the stable release is available. Issues are welcome
+for unexpected changes not noted below or in the changelog.
+
+## 🚨 Breaking changes
+
+While `lit-html` 2.0 is intended to be a backward-compatible change for the
+majority of 1.x users, please be aware of the following notable breaking
+changes, and see the full [changelog](CHANGELOG.md#200---unreleased) for more
+details:
+  * New `directive` and `Part` API (see below for migration info)
+  * `render()` no longer clears its container on first render
+  * Custom `templateFactory`, `TemplateProcessor`, and custom tag functions are no
+    longer supported
+
+## 🚨 Known issues/limitations
+
+* **Browser support**: This pre-release should run on modern browsers, however a
+  change to factor legacy browser support (IE11, etc.) into an opt-in package is
+  ongoing. As such, this release will not run on some older browsers. This is a
+  temporary state.
+* **Limited directive implementation**: The following directives are not yet
+  implemented. This is a temporary state:
+  * `asyncAppend`
+  * `asyncReplace`
+  * `templateContent`
+  * `unsafeSvg`
+  * `until`
+
+## 🚨 Migrating directives
+
+While the API for _using_ directives should be 100% backward-compatible with
+1.x, there is a breaking change to how custom directives are _authored_. The API
+change improves ergonomics around making stateful directives while providing a
+clear pattern for SSR-compatible directives: only `render` will be called on the
+server, while `update` will not be.
+
+<details>
+<summary>Expand here for details on migrating directives.</summary>
+
+### Overview of directive API changes
+| | 1.x API | 2.0 API |
+|-|-----|-----|
+| Code idiom for directive | function that takes directive arguments, and returns function that takes `part` and returns value | class with `update` & `render` methods which accept directive arguments |
+| Where to do declarative rendering | pass value to `part.setValue()` | return value from `render()` method |
+| Where to do imperative DOM/part manipulation | directive function | `update()` method |
+| Where state is stored between renders | `WeakMap` keyed on `part` | class instance fields |
+| How part validation is done | `instanceof` check on `part` in every render | `part.type` check in constructor
+
+### Example directive migration
+Below is an example of a fictional directive, and how to migrate it to the new
```

Comment:
```suggestion
Below is an example of a functional directive, and how to migrate it to the new
```

Reply:
That was actually intended to be _fiction_al... (i.e. aka it's made up, not a useful thing). Is that a weird way to say it?

Reply:
"fictional functional"? :)

Reply:
More seriously, "example of a lit-html 1.x directive"